lib/evmone/tracing.cpp

```diff
@@ -67,6 +68,48 @@ class HistogramTracer : public BaseTracer
         BaseTracer::on_execution_end(result);
     }
 
+public:
+    using BaseTracer::BaseTracer;
+};
+
+class InstructionTracer : public BaseTracer
+{
+    const char* const* m_opcode_names = nullptr;
+
+    void on_execution_start(
+        evmc_revision rev, const evmc_message& msg, bytes_view code) noexcept override
+    {
+        BaseTracer::on_execution_start(rev, msg, code);
+        m_opcode_names = evmc_get_instruction_names_table(rev);
+        m_out << std::dec;  // Set number formatting to dec, JSON does not support other forms.
+
+        m_out << "{";
+        m_out << R"("start":true)";
+        m_out << R"(,"depth":)" << msg.depth;
```

Comment:
Would be nice displaying the startgas, `msg.gas_limit` should have it ?

Reply:
Done.